notebooks/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,660 @@
+    "          window=365,\n",
+    "          center=True,\n",
+    "          min_periods=183,\n",
+    "      ).mean()\n",
+    "      ax = average_sales.plot(alpha=0)\n",
+    "      ax = trend.plot(ax=ax, linewidth=3)\n",
+    "  plt.title(family_value)\n",
+    "  plt.legend(train_data_copy['type'].unique())\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>As we can see, the store type has a great impact on the target (type D is leader and it is logical, because the plot above shows us, that type D stores are the most), that is why 'type' feature is very important.</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(7,5))\n",
+    "sns.countplot(x=train_data['cluster'], alpha=0.7, data=train_data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "sales_and_cluster_data = train_data_copy.drop(['date', 'family', 'onpromotion', 'city', 'type'], axis=1)\n",
+    "sales_and_cluster_data.corr()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figsize=(10, 7)\n",
+    "plt.scatter(x='cluster', y='sales', data=sales_and_cluster_data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>'cluster' feature doesn't correlate with the target, and I can't see any dependecies between these 2 features. But I think that this feature can be useful, because it connects similar stores together.</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/",
+     "height": 206
+    },
+    "id": "Tiavh7PK-HYv",
+    "outputId": "4d198de7-7161-4413-eefa-45b6201bd547"
+   },
+   "outputs": [],
+   "source": [
+    "oil_data = pd.read_csv('C:\\\\Users\\\\Owner\\\\Desktop\\\\Proga\\\\REPOS\\\\mentorship_EPAM\\\\data\\\\oil.csv')\n",
+    "oil_data[\"time\"] = np.arange(len(oil_data.index))\n",
+    "oil_data.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "FDbzS7RpNnKK",
+    "outputId": "ef457345-a324-40da-db3e-6a7362248416"
+   },
+   "outputs": [],
+   "source": [
+    "oil_data.info()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/",
+     "height": 297
+    },
+    "id": "Bt59gF6ePlOZ",
+    "outputId": "9af1e0e1-f125-40ef-ba81-c9c889b7ba91"
+   },
+   "outputs": [],
+   "source": [
+    "oil_data_copy = oil_data.copy()\n",
+    "oil_data_copy.plot(kind=\"scatter\", x=\"time\", y=\"dcoilwtico\")"
```

Comment:
Ordinary plot instead of a scatter one will be better here, I suppose. And probably some weekly or monthly rolling aggregations will help better undestand trends.